src/runtime/auto-prompt-manager.js

```diff
@@ -180,45 +204,48 @@ export class AutoPromptManager {
     dismissedPrompts,
     params
   ) {
+    let shouldShowAutoPrompt;
     return this.shouldShowAutoPrompt_(
       clientConfig,
       entitlements,
       params.autoPromptType
-    ).then((shouldShowAutoPrompt) => {
-      const potentialActionPromptType = this.getAudienceActionPromptType_({
-        article,
-        autoPromptType: params.autoPromptType,
-        dismissedPrompts,
-        shouldShowAutoPrompt,
-      });
-      const promptFn = potentialActionPromptType
-        ? this.audienceActionPrompt_({
-            action: potentialActionPromptType,
-            autoPromptType: params.autoPromptType,
-          })
-        : params.displayLargePromptFn;
-
-      if (!shouldShowAutoPrompt) {
-        if (
-          this.shouldShowBlockingPrompt_(
-            entitlements,
-            /* hasPotentialAudienceAction */ !!potentialActionPromptType
-          ) &&
-          promptFn
-        ) {
-          promptFn();
+    )
+      .then((shouldShowAutoPrompt_) => {
+        shouldShowAutoPrompt = shouldShowAutoPrompt_;
+        return this.getAudienceActionPromptType_({
```

Comment:
This has the issue of passing the promise object, rather than resolving it. I instead refactored to use `Promise.all` to pass an array of `potentialActionPromptType` and `shouldShowAutoPrompt` so that the local variable is no longer used; how does this look to you?

Reply:
Looks better! Thanks :)